tket/src/Utils/MatrixAnalysis.cpp

```diff
@@ -292,173 +292,55 @@ Eigen::VectorXcd apply_qubit_permutation(
   return perm_m * v;
 }
 
+// How close TK2(a,b,c) is from the identity
+double trace_fidelity(double a, double b, double c) {
+  constexpr double g = PI / 2;
+  a *= g;
+  b *= g;
+  c *= g;
+  double trace_sq = 16. * (pow(cos(a) * cos(b) * cos(c), 2) +
+                           pow(sin(a) * sin(b) * sin(c), 2));
+  return (4. + trace_sq) / 20.;
```

Comment:
Bit confused by this. The result is always at least 0.2, even when a=b=c=0, which equates to the identity?
And if say cos(a)=0 and sin(b)=0, the result is exactly 0.2, independent of c -- which also seems wrong.

Reply:
This formula comes straight from https://arxiv.org/pdf/1811.12926.pdf. See equations B3 and B5.

Regarding your two specific examples:

- If `a=b=c=0`, then `cos(a)cos(b)cos(c) = 1` and hence the fidelity is `1.0` as it should be.
- `a, b, c` must be in the Weyl chamber. `sin(b) = 0` ⇒ `b = 0` ⇒ `c = 0`. Hence the fidelity doesn't need to depend on `c`.
 
I have added a comment specifying that a,b,c must be in the Weyl chamber. Is there anything else that I am missing?

Reply:
You're right, I was confused by trying to read it as distance from the identity, which is what the comment above makes it sound like. Perhaps that comment could be clarified?

Reply:
Yes, I've removed that comment. I've expanded the doxygen docs in the `hpp`.